Task3/Task3_Jordan_Petrov/src/python/car_wash/pool.py

```diff
@@ -0,0 +1,121 @@
+"""
+Generic db pooling code for two types of db users:
+ - read and write,
+ - read only
+You can use them as context manager
+You can run 'python3 pool.py -h' see how to parse parameters from console
+"""
+import argparse
+import logging
+
+import psycopg2
+
+from config import READ_ONLY_USER, READ_WRITE_USER
+from utils import SingletonMeta
+
+logging.basicConfig(
+    filename='../../../pool.log',
+    format='%(asctime)s - %(message)s',
+    level=logging.INFO
+)
+
+
+class BaseConnectionPool(metaclass=SingletonMeta):
+    """
+    Base class for the read only and read write classes
+    """
+
+    def __init__(self, conns: int, *args, **kwargs):
```

Comment:
Use *args / **kwarks is possible but it is a bad practice. Please use variables.

Reply:
Also, Please put conns into config.py